### ISSUE TITLE
Add way to add comments in code snippets. 

### ISSUE THREAD

**Issue by nainar** [ampproject/amp-by-example#1266]
E.g. in the case of `<amp-fx="fade-in">` (https://ampbyexample.com/components/amp-fx-collection/) it would be useful to add a code comment. However, because of the generator the comments break functionality. Is it possible to add comments to the code snippets as well?

**Comment by sebastianbenz** [comment]
What do you mean by code comment? Can you provide an example of what you want to accomplish.

**Comment by nainar** [comment]
Given the following written up in an html page in ABE:
```
  <!--
Sentence 1
  -->
  <div class="header">
  </div>
```
`Sentence 1` is converted into text displayed on the screen and the  divs are in encased in a code Example box. However, if we wanted to add comments to that code example box as follows:
```
  <!--
Sentence 1
  -->
  <div class="header"> <!-- This div has a class header -->
  </div>
```
This would create a text with `Sentence 1`, followed by a code example block with `<div class="header">` followed by another text with `This div has a class header `, followed by a code example block with a `</div>`. The intention is to have a text with `Sentence 1`, followed by a code example block with:
```
  <div class="header"> <!-- This div has a class header -->
  </div>
```

This isn't possible due to the way the comments are converted into text boxes. Does this make it clear?

**Comment by sebastianbenz** [comment]
This has been fixed by #1338 thanks to @fstanis.
